### PR TITLE
Pub 2089 - Align JOH functionality

### PR DESCRIPTION
**Pull request by NatashaAlker**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/PUB-2089

### Change description ###

Update pdfs and email summaries of impacted lists to align the way judiciary is displayed

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/pip/channel/management/services/helpers/JudiciaryHelper.java [hunk 1]
```diff
@@ -12,49 +12,37 @@ private JudiciaryHelper() {
     }
 
     public static String findAndManipulateJudiciary(JsonNode judiciaryNode) {
-        AtomicReference<StringBuilder> formattedJudiciary = new AtomicReference<>(new StringBuilder());
```
Comment (KianKwa) [inline src/main/java/uk/gov/hmcts/reform/pip/channel/management/services/helpers/JudiciaryHelper.java:15]: I think the code could be simplified. It keeps checking for non-empty string and appends comma to every element then trim it off later.  I think it is cleaner to handle empty string and delimiter at the end. You can try doing something like below:

```
AtomicReference<String> presidingJudiciary = new AtomicReference<>("");
List<String> judiciaries = new ArrayList<>();

if (judiciaryNode.has(JUDICIARY)) {
  judiciaryNode.get(JUDICIARY).forEach(judiciary -> {
      String johKnownAs = GeneralHelper.findAndReturnNodeText(judiciary, "johKnownAs");
      if ("true".equals(GeneralHelper.findAndReturnNodeText(judiciary, "isPresiding"))) {
          presidingJudiciary.set(johKnownAs);
      } else {
          judiciaries.add(johKnownAs);
      }
  });
}
judiciaries.add(0, presidingJudiciary.get());
judiciaries.stream()
  .filter(StringUtils::isNotBlank)
  .collect(Collectors.joining(", "));
```

File: src/main/java/uk/gov/hmcts/reform/pip/channel/management/services/filegeneration/IacDailyListFileConverter.java [hunk 2]
```diff
@@ -58,7 +58,7 @@ private void calculateListData(JsonNode artefact) {
 
             courtList.get("courtHouse").get("courtRoom").forEach(
                 courtRoom -> courtRoom.get("session").forEach(session -> {
-                    String formattedJoh = JudiciaryHelper.findAndManipulateJudiciaryForIac(session);
+                    String formattedJoh = JudiciaryHelper.findAndManipulateJudiciary(session);
```
Comment (KianKwa) [inline src/main/java/uk/gov/hmcts/reform/pip/channel/management/services/filegeneration/IacDailyListFileConverter.java:61]: I just want to double check we should handle judiciary the same way for IAC. IAC wasn't mentioned in the ticket. Also I been to the meeting about the IAC list with other teams and agreeing IAC should display judiciary title and surname as it conforms with the initial requirements from McGirr. Just adding a comment here. I will check with Chris.

Comment (NatashaAlker) [reply src/main/java/uk/gov/hmcts/reform/pip/channel/management/services/filegeneration/IacDailyListFileConverter.java]: You're right, I checked with Chris and both the IAC List and Magistrates Standard List need to keep using the old way :)

File: src/main/java/uk/gov/hmcts/reform/pip/channel/management/services/helpers/JudiciaryHelper.java [hunk 3]
```diff
@@ -12,51 +14,48 @@ private JudiciaryHelper() {
...
                 }
             });
         }
 
-        return GeneralHelper.trimAnyCharacterFromStringEnd(formattedJudiciary.toString());
+        if (StringUtils.isNotBlank(presidingJudiciary.get())) {
+            judiciaries.add(0, String.valueOf(presidingJudiciary.get()));
+        }
+
+        return String.join(", ", judiciaries);
```
Comment (KianKwa) [inline src/main/java/uk/gov/hmcts/reform/pip/channel/management/services/helpers/JudiciaryHelper.java:35]: Some of the non-presiding judiciary could be blank too if the JohKnownAs value is an empty string. If you do it this way, you can potentially includes the blank value in the comma delimited output string. It is better to do the blank check at the end for all presiding and non-presiding judiciaries.

judiciaries.add(0, presidingJudiciary.get());
judiciaries.stream()
  .filter(StringUtils::isNotBlank)
  .collect(Collectors.joining(", "));

File: src/main/java/uk/gov/hmcts/reform/pip/channel/management/services/helpers/JudiciaryHelper.java [hunk 4]
```diff
@@ -12,51 +14,48 @@ private JudiciaryHelper() {
...
-        String johKnownAs = GeneralHelper.findAndReturnNodeText(judiciary, "johKnownAs");
-        if (StringUtils.isNotBlank(johKnownAs)) {
-            formattedJudiciary
-                .append(johKnownAs)
-                .append(", ");
+        if (StringUtils.isNotBlank(presidingJudiciary.get())) {
+            judiciaries.add(0, String.valueOf(presidingJudiciary.get()));
         }
+
+        return String.join(", ", judiciaries);
```
Comment (KianKwa) [inline src/main/java/uk/gov/hmcts/reform/pip/channel/management/services/helpers/JudiciaryHelper.java:59]: Same comment like the one in findAndManipulateJudiciary()

File: src/main/java/uk/gov/hmcts/reform/pip/channel/management/services/helpers/JudiciaryHelper.java [hunk 5]
```diff
@@ -12,51 +14,48 @@ private JudiciaryHelper() {
...
 
-        try {
-            session.get(JUDICIARY).forEach(judiciary -> {
-                if (formattedJudiciary.length() != 0) {
-                    formattedJudiciary.append(", ");
+        if (judiciaryNode.has(JUDICIARY)) {
+            judiciaryNode.get(JUDICIARY).forEach(judiciary -> {
+                String johTitle = GeneralHelper.findAndReturnNodeText(judiciary, "johTitle");
+                String johNameSurname = GeneralHelper.findAndReturnNodeText(judiciary, "johNameSurname");
+                String judgeName = johTitle + " " + johNameSurname;
```
Comment (KianKwa) [inline src/main/java/uk/gov/hmcts/reform/pip/channel/management/services/helpers/JudiciaryHelper.java:46]: If the johTitle is blank, you will end up with an unwanted extra space in front of the surname. It is better to use:

Stream.of(johTitle, johNameSurname)
  .filter(StringUtils::isNotBlank)
  .collect(Collectors.joining(" "));

**Review by KianKwa — APPROVED** [review]

**Review by ChrisS1512 — APPROVED** [review]

**Pull request merged**